crates/storage/errors/src/writer.rs

```diff
@@ -2,15 +2,36 @@ use crate::db::DatabaseError;
 use reth_primitives::StaticFileSegment;
 
 /// `UnifiedStorageWriter` related errors
-#[derive(Clone, Debug, thiserror_no_std::Error, PartialEq, Eq)]
+/// `StorageWriter` related errors
+#[derive(Clone, Debug, derive_more::Display, PartialEq, Eq)]
 pub enum UnifiedStorageWriterError {
+    /// Database writer is missing
+    #[display(fmt = "Database writer is missing")]
+    MissingDatabaseWriter,
     /// Static file writer is missing
-    #[error("Static file writer is missing")]
+    #[display(fmt = "Static file writer is missing")]
     MissingStaticFileWriter,
     /// Static file writer is of wrong segment
-    #[error("Static file writer is of wrong segment: got {0}, expected {1}")]
+    #[display(fmt = "Static file writer is of wrong segment: got {_0}, expected {_1}")]
     IncorrectStaticFileWriter(StaticFileSegment, StaticFileSegment),
     /// Database-related errors.
-    #[error(transparent)]
-    Database(#[from] DatabaseError),
+    Database(DatabaseError),
 }
+
+#[cfg(feature = "std")]
+impl std::error::Error for UnifiedStorageWriterError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+       match self {
+            Self::Database(source) => {
+                std::error::Error::source(source)
+            },
+           _ => Option::None
+       } 
+    }
+}
+
+impl From<DatabaseError> for UnifiedStorageWriterError {
+    fn from(value: DatabaseError) -> Self {
+        Self::Database(value)
+    }
+}
```

Comment:
this can be derived

Reply:
let's leave it as a manual impl as we did in other PRs